runtime/doc/eval.txt

```diff
@@ -5449,6 +5449,30 @@ max({list})	Return the maximum value of all items in {list}.
 		be used as a Number this results in an error.
 		An empty |List| results in zero.
 
+menu_get({path}, {modes})				*dump_menu()*
+		Returns a |Dictionary| with all the submenu of {path} (set to 
+		an empty string to match all menus). Only the commands matching {modes} are 
+		returned ('a' for all, 'i' for insert see |creating-menus|).
+
+    For instance, executing:
+<
+    nnoremenu &Test.Test inormal
+    inoremenu Test.Test insert
+    vnoremenu Test.Test x
+    echo menu_get("")
+>
+should produce:
+<
+[
+{'hidden': 0, 'submenus':
+  [{'hidden': 0, 'name': 'Test', 'priority': 500, 'shortcut': 0, 'implementations':
+ {'s': {'noremap': 1, 'enabled': 1, 'silent': 0, 'sid': 1, 'rhs': 'x'}, 'v': {'noremap': 1, 'enabled': 1, 's
+ilent': 0, 'sid': 1, 'rhs': 'x'}, 'i': {'noremap': 1, 'enabled': 1, 'silent': 0, 'sid': 1, 'rhs': 'insert'},
+ 'n': {'noremap': 1, 'enabled': 1, 'silent': 0, 'sid': 1, 'rhs': 'inormal'}}}], 'name': 'Test', 'priority':
+500, 'shortcut': 84}
+]
```

Comment:
Heh. Sadly `gqq` can't format Vim's own syntax. But I would:

1. `J`oin it to one line
2. `json_encode()` it: `:put =json_encode(<C-R>0)`
3. Format it:
    ```
    :setl formatprg=python\ -m\ json.tool
    gqq
    ```

(If it's just showing a generic map, doesn't matter if it's JSON or VimL, I suppose)